homeassistant/components/recorder/migration.py

```diff
@@ -363,6 +363,13 @@ def _apply_update(engine, new_version, old_version):
         if engine.dialect.name == "mysql":
             _modify_columns(engine, "events", ["event_data LONGTEXT"])
             _modify_columns(engine, "states", ["attributes LONGTEXT"])
+    elif new_version == 13:
+        if engine.dialect.name == "mysql":
+            _modify_columns(engine, "events", ["time_fired DATETIME(6)"])
+            _modify_columns(engine, "events", ["created DATETIME(6)"])
+            _modify_columns(engine, "states", ["last_changed DATETIME(6)"])
+            _modify_columns(engine, "states", ["last_updated DATETIME(6)"])
+            _modify_columns(engine, "states", ["created DATETIME(6)"])
```

Comment:
`modify_columns` can take multiple modifications at once. (this will need black)
```suggestion
            _modify_columns(engine, "events", ["time_fired DATETIME(6)", "created DATETIME(6)"])
            _modify_columns(engine, "states", ["last_changed DATETIME(6)", "last_updated DATETIME(6)", "created DATETIME(6)"])
```

Reply:
Oh right, good point. That will make the migration perform much better too.